### ISSUE TITLE
Electron-updater cache dir pending not cleaned up.

### ISSUE THREAD

**Issue by Haradrim** [electron-userland/electron-builder#6269]
* **Electron-Builder Version**: 20.44.4

* **Node Version**: 14.17.0

* **Electron Version**: 3.1.13
* **Electron Type (current, beta, nightly)**: /

* **Target**:  NSIS

Auto update does not work because the pending folder is not cleaned up after a successful update, causing the next update to fail.

I see following information in the logs:
```
[2021-09-20 11:43:37.334] [info] App starting...
[2021-09-20 11:43:39.419] [info] Checking for update
[2021-09-20 11:43:39.723] [info] Found version 12.0.48 (url: MY APP Setup 12.0.48.exe)
[2021-09-20 11:43:39.739] [info] Downloading update from MY APP Setup 12.0.48.exe
[2021-09-20 11:43:39.761] [error] Error: Error: EEXIST: file already exists, mkdir 'C:\Users\dlwsa\AppData\Local\my-app-updater\pending'
[2021-09-20 11:43:39.775] [info] Error in auto-updater: Error: EEXIST: file already exists, mkdir 'C:\Users\dlwsa\AppData\Local\my-app-updater\pending'
```
For context, the app starting here is a previous version which in turn was auto updated successfully from a manual install. The update fails and does not prompt to install the new version.

**Comment by mmaietta** [comment]
Can you try a more recent version of electron-builder and electron-updater please? I see you reported for 20.44.4

**Comment by Haradrim** [comment]
Apologies, tried again with the latest version of electron-builder, electron-updater was already on the latest version. Same issue persists. Update is detected by electron-update, but fails because the pending folder already exists.

- electron-updater: 4.3.9
- electron-builder: 22.11.7

Is there anything more I can provide from my side?

**Comment by mmaietta** [comment]
@Haradrim , would you be willing to open up a PR to resolve that? Setting up a local dev environment is fairly simple :) 
https://github.com/electron-userland/electron-builder/blob/master/CONTRIBUTING.md#to-setup-a-local-dev-environment

**Comment by vukhacbiet** [comment]
> @mmaietta Temporary resolved the issue by downgrading to version 4.2.0 of electron updater. Seems to be an issue with fs-extra 10.0. The function emptyDir does not work as intended.

Do you resolve it?

**Comment by lxmeetlx** [comment]
> @mmaietta Temporary resolved the issue by downgrading to version 4.2.0 of electron updater. Seems to be an issue with fs-extra 10.0. The function emptyDir does not work as intended.

Did you solve this issue?
